Model/BalancepayMethod.php

```diff
@@ -389,11 +397,13 @@ public function capture(InfoInterface $payment, $amount)
             $balanceVendorId = null;
 
             foreach ($orderItems as $item) {
-                $balanceVendorId = $helper->getBalanceVendors($item->getProductId());
+                $_balanceVendorId = $this->helper->getBalanceVendors($item->getProductId());
```

Comment:
what is the difference between `$_balanceVendorId` and `$balanceVendorId`? please try to come up with more descriptive variable name. 

Also, you are setting `$balanceVendorId` to null, so I think the condition statement on line 411 will always evaluate to false:
`$balanceVendorId && $balanceVendorId !== $_balanceVendorId`

Reply:
These variables were actually used in the previous module, I kept them as it is and added new logic around them. 
I am going to update those variables and give more descriptive name, Its confusing as well to understand the logic.